vshaxe/hashlink-debugger
# Add debugging support for MacOS

This PR adds debugging support for OSX/MacOS. Things done:

- [x] Implemented `NodeDebugApiMac` which uses NodeFFI to call functions from a shared lib ~~`libhldebug.dylib`~~ `libhl.dylib` from Hashlink.
- [x] Fixed two issues in ``Debugger.hx``
- [x] ~~Add `entitlements.xml` and a codesign target to `Makefile`; It's required to codesign the `hl` executable in order to get debugging permissions on MacOS.~~`entitlements.xml` moved to Hashlink project
- [x] ~~Added documentation on signing to `MacOS.md` and referenced it from the `README.md`~~
- [x] Added a guard to `Extension.hx` which checks if ~~the `libhldebug.dylib` is available (meaning that the Hashlink version is supported)~~ `hl` has a compatible version number and is properly codesigned

**Known Issues**
- [ ] There still is an issue with the "Stop on all exceptions" capability, which I therefore excluded from the Mac version for now

Feel free to comment and **please help test it!**.

**How to Test**
1. ~Pull the Hashlink version from PR: https://github.com/HaxeFoundation/hashlink/pull/367 and install it~ Pull the latest master branch of Hashlink from Github 
2. ~Follow instructions in `MacOS.md` for code-signing~ Make sure you run `make codesign_osx` during Hashlink installation prior to `make install`
3. Pull the Hashlink-Debugger version from **this** PR and build it (``make deps``)
4. Remove the marketplace version of hashlink-debugger from your VSCode extensions (_~/.vscode/extensions_) folder
5. Make a symlink from inside your VSCode extension folder to the version you pulled
6. Restart VSCode and start a debugging configuration
